src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -63,30 +70,10 @@ private Value GetToolInfo(string toolName, string toolFileVersion, Version toolA
             var builder = ArrayBuilder<KeyValuePair<string, Value>>.GetInstance();
             builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.ToolName, toolName));
             builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.ToolAssemblyVersion, toolAssemblyVersion.ToString(fieldCount: 3)));
-            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.ToolFileVersion, GetToolFileVersionSubStr(toolFileVersion)));
+            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.ToolFileVersion, toolFileVersion));
             return Value.Create(builder.ToImmutableAndFree(), this);
         }
 
-        private string GetToolFileVersionSubStr(string toolFileVersion)
-        {
-            // Our log format specifies that fileVersion can have at most 3 fields.
```

Comment:
Note that this was used for the `fileVersion` not the `version` property. The `fileVersion` example in the spec shows four parts there and the text says to get it from the operating system, which is where we got it (via FileVersionInfo). That's why I reasoned that we should not be striping any part of it here.

As for using the full semantic version in  `version`, I don't see a way to get it the prerelease part, but perhaps I missed it. Without that, we continue to use AssemblyVersion.ToString(fieldCount: 3) which will have 3 parts. It is a valid server, but will appear stable when the compiler is prerelease. I can open an issue to track having the semantic version that corresponds to the nuget package...